source/TimePicker.js

```diff
@@ -585,14 +608,27 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
+			var meridiems = this.meridiems,
+				hour = this.$.hour.getValue(),
+				oldIndex = event.old,
+				newIndex = this.$.meridiem.getValue(),
+				offset, start, end, newHour;
 
-			if (this.value) {
-				// value is 0 for am, 1 for pm
-				// reset the hour to < 12 and then add 12 if it's pm
-				hour = hour%12 + value*12;
-				this.updateHours(hour);
+			offset = hour - parseInt(meridiems[oldIndex]['start'], 10);
```

Comment:
Yes, the Chinese meridiems are like the Ethiopian ones, should we choose to support them. There are other ones like that as well, but ilib doesn't support them yet.

Ilib has a call to retrieve all the meridiems and their start/end times: DateFmt.DateFmt.getMeridiemsRange() which returns an array of {name:string, start:string, end:string} objects which give the details of what the meridiems are called, when they start in "hours:minutes" format, and when they end, also in "hours:minutes". If you only key off of that, then you can support any meridiems style that ilib supports.